pkg/operator/ceph/csi/spec.go

```diff
@@ -213,6 +219,25 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	} else {
 		tp.ForceCephFSKernelClient = "true"
 	}
+
+	// "--leader-election-type=leases" parameter in external-attacher is
+	// removed in v2.1.0 but it is required if the external-attacher version is
+	// v1.2.x
+	attacher := strings.Split(CSIParam.AttacherImage, ":")
+	if len(attacher) > 1 {
+		if strings.HasPrefix(attacher[1], "v1.2.") {
+			tp.SetAttacherLeaderElectionType = true
+		}
+	}
+
+	csiPluginImage := strings.Split(CSIParam.CSIPluginImage, ":")
+	// as ceph-csi v2.x.x support resizer, enable it
+	if len(csiPluginImage) > 1 {
+		if strings.HasPrefix(csiPluginImage[1], "v2.") {
```

Comment:
Looking at the image label is not very reliable, but it will have to do for now. We'll need #4256 for more reliable csi version detection.

Reply:
Yes, thats what I was thinking, once #4256 is merged we can use same mechanism here also

Reply:
I don't think we will need to backport #4256 to the release-1.2 branch, so I would recommend we don't need to worry about updating this later. It can stay this way in the release-1.2 branch.